gapic/src/main/com/google/gapid/perfetto/models/FrameEventsTrack.java

```diff
@@ -333,6 +388,62 @@ public void getRange(Consumer<TimeSpan> span) {
         slice.getRange(span);
       }
     }
+
+    public FrameSelection getSelection() {
+      FrameSelection selection = new FrameSelection();
+      slices.forEach(s -> selection.combine(s.getSelection()));
+      return selection;
+    }
+  }
+
+  public static class FrameSelection {
+    public final List<Long> frameNumbers;
+    public final List<String> layerNames;
+    public static FrameSelection EMPTY = new FrameSelection();
+
+    public FrameSelection() {
+      frameNumbers = Lists.newArrayList();
+      layerNames = Lists.newArrayList();
+    }
+
+    public FrameSelection(long[] frameNumbers, String[] layerNames) {
+      this.frameNumbers = Lists.newArrayList(Longs.asList(frameNumbers));
+      this.layerNames = Lists.newArrayList(layerNames);
+    }
+
+    public boolean contains(long[] f, String[] l) {
+      for (int i = 0; i < f.length; i++) {
```

Comment:
Just a note that this function will also be queried every time rendering the track, and will be called for every visible slices. Is there any better key to identify the selection, and make this function run faster? If not then it's fine to keep it this way.

Reply:
Leaving it like this for now, based on our discussion.